### ISSUE TITLE
[ENHANCEMENT] allow myloader to also read from .bz2 files, via a pipe filter

### ISSUE THREAD

**Issue by technomagos** [mydumper/mydumper#513]
*Describe the enhancement*

We use myloader to backup large maria databases. At the end of non-compressed backup, we run parallel bzip2 compression (via lbbzip2 which uses multiple threads, in similar spirit as mydumper), which guevs higher compression than gzip/zstd, which is supproted. The backup files are so stored in .bz2 format which are placed in offline  storage. The (smaller) schema file per database is kept in a .sql file, for easy viewing.

The request is for myloader to be able to process .bz2 files directly to do the restore .  A simple idea for that could be that you add a --file-filter flag , which processes any file that is in .bz2 format, or similar, in a pipe. Non compressed (.sql) file would be matched without issue.

Currently we need to decompress all files first from .bz2  format into a temp directory so we need to 3-5x more storage to perform the restore which should not be required.

**Comment by davidducos** [comment]
Hi @technomagos, doesn't stream helps you to accomplish the task?

**Comment by technomagos** [comment]
I assume they way streaming works (sorry if i make a mistake, it is a very new feature), i will need to save one large file for all tables in a database backup. So, will not be able anymore to restore selectively, some of the tables from the mydumper backup, which i can do now as they are stored on separate files on disk (but need to bzip2-decompress first).

**Comment by davidducos** [comment]
Hi @technomagos, yes, steaming backs up is really new in MyDumper!! No need to apologize. 

For zstd, we used a gzip wrapper library. Do you know if we can use some similar implementation for bz2?

**Comment by davidducos** [comment]
@technomagos Sorry, I didn't know that we have a libbz2-devel package in Ubuntu, we might be able to include the lib and point to the right read and write functions. I will be investigating this next week. I will keep you posted!

**Comment by davidducos** [comment]
Hi @technomagos,
As we implemented the --exec functionality in #709, you are going to be able to do this:
``` 
[root@localhost mydumper]# rm -rf data ; ./mydumper -B myd_test -o data --no-locks --exec "/usr/bin/bzip2 FILENAME"
[root@localhost mydumper]# ls data/
metadata.bz2                        myd_test.mydumper_2-schema.sql.bz2  myd_test.mydumper_5-schema.sql.bz2   myd_test.pertest_child-schema.sql.bz2        myd_test.table_view_2_b-metadata.bz2
myd_test.mydumper_0.00000.sql.bz2   myd_test.mydumper_3.00000.sql.bz2   myd_test.mydumper_6.00000.sql.bz2    myd_test-schema-create.sql.bz2               myd_test.table_view_2_b-schema.sql.bz2
myd_test.mydumper_0-metadata.bz2    myd_test.mydumper_3-metadata.bz2    myd_test.mydumper_6-metadata.bz2     myd_test.t.00000.sql.bz2                     myd_test.t-metadata.bz2
myd_test.mydumper_0-schema.sql.bz2  myd_test.mydumper_3-schema.sql.bz2  myd_test.mydumper_6-schema.sql.bz2   myd_test.table_view_1_a-metadata.bz2         myd_test.t-schema.sql.bz2
myd_test.mydumper_1.00000.sql.bz2   myd_test.mydumper_4.00000.sql.bz2   myd_test.perftest_2-metadata.bz2     myd_test.table_view_1_a-schema.sql.bz2       myd_test.t_w_g.00000.sql.bz2
myd_test.mydumper_1-metadata.bz2    myd_test.mydumper_4-metadata.bz2    myd_test.perftest_2-schema.sql.bz2   myd_test.table_view_1_b-schema.sql.bz2       myd_test.t_w_g-metadata.bz2
myd_test.mydumper_1-schema.sql.bz2  myd_test.mydumper_4-schema.sql.bz2  myd_test.perftest-metadata.bz2       myd_test.table_view_1_b-schema-view.sql.bz2  myd_test.t_w_g-schema.sql.bz2
myd_test.mydumper_2.00000.sql.bz2   myd_test.mydumper_5.00000.sql.bz2   myd_test.perftest-schema.sql.bz2     myd_test.table_view_2_a-schema.sql.bz2       myd_test.v-schema.sql.bz2
myd_test.mydumper_2-metadata.bz2    myd_test.mydumper_5-metadata.bz2    myd_test.pertest_child-metadata.bz2  myd_test.table_view_2_a-schema-view.sql.bz2  myd_test.v-schema-view.sql.bz2
```
